client/block.c

```diff
@@ -353,7 +378,22 @@ static int valid_signature(const struct xdag_block *b, int signo_r, int keysLeng
 	if (signo_s >= 0) {
 		for (i = 0; i < keysLength; ++i) {
 			hash_for_signature(buf, keys + i, hash);
+#if OPENSSL == 1
+			if (!xdag_verify_signature(keys[i].key, hash, b->field[signo_r].data, b->field[signo_s].data)) {
+#elif OPENSSL == 0
+			if (!xdag_verify_signature_noopenssl(keys[i].pub, hash, b->field[signo_r].data, b->field[signo_s].data)) {
+#elif OPENSSL == 2
+			int res1=0,res2=0;
+			res1=!xdag_verify_signature_noopenssl(keys[i].pub, hash, b->field[signo_r].data, b->field[signo_s].data);
+			res2=!xdag_verify_signature(keys[i].key, hash, b->field[signo_r].data, b->field[signo_s].data);
+			if (res1!=res2){
+				xdag_debug("Different result between openssl and secp256k1: res openssl=%2d res secp256k1=%2d key parity bit = %ld key=[%s] hash=[%s] r=[%s], s=[%s]", res2, res1, ((uintptr_t)keys[i].pub & 1), 
+                                                                        xdag_log_hash((uint64_t*)((uintptr_t)keys[i].pub & ~1l)) , xdag_log_hash(hash), xdag_log_hash(b->field[signo_r].data), xdag_log_hash(b->field[signo_s].data));
+			}
+			if(res2){
+#else
 			if (!xdag_verify_signature(keys[i].key, hash, b->field[signo_r].data, b->field[signo_s].data)) {
```

Comment:
What is the difference between this line and line under #if OPENSSL == 1 ?

Reply:
there isn't, it's just a default version

Reply:
So, it is better to avoid duplication and leave conditions:
#if OPENSSL == 0
...
#elif OPENSSL == 2
...
#else
...
#endif

Reply:
done